CMakeLists.txt

```diff
@@ -145,6 +148,21 @@ if(CELERITAS_USE_VecGeom)
   if(NOT VecGeom_GDML_FOUND)
     message(SEND_ERROR "VecGeom GDML capability is required for Celeritas")
   endif()
+
+  set_target_properties(VecGeom::vecgeom PROPERTIES
+    CELERITAS_CUDA_STATIC_LIBRARY VecGeom::vecgeomcuda_static
+    CELERITAS_CUDA_SHARED_LIBRARY VecGeom::vecgeomcuda
+    CELERITAS_CUDA_FINAL_LIBRARY VecGeom::vecgeomcuda
+  )
+  set_target_properties(VecGeom::vecgeomcuda PROPERTIES
+    CELERITAS_CUDA_LIBRARY_TYPE Shared
+    CELERITAS_CUDA_STATIC_LIBRARY VecGeom::vecgeomcuda_static
+  )
+  set_target_properties(VecGeom::vecgeomcuda_static PROPERTIES
+    CELERITAS_CUDA_LIBRARY_TYPE Static
+    CELERITAS_CUDA_SHARED_LIBRARY VecGeom::vecgeomcuda
+    CELERITAS_CUDA_FINAL_LIBRARY VecGeom::vecgeomcuda
+  )
```

Comment:
I think we should plan on propagating the new "modern cmake" RDC linkage code back to vecgeom (or even to veccore etc. if those ever generate CUDA code). If you and @drbenmorgan agree, we might want to proactively scope these propeties as `VECGEOM_` ratherr than `CELERITAS_`

Reply:
Agreed on this - though I'd like to review the process/changes first just to understand them and check the "usage requirement" propagation. That's best done once this is merged with the final implementation, so will keep an eye for that.